src/mongocxx/exception/base.cpp

```diff
@@ -14,21 +14,26 @@
 
 #include <mongocxx/exception/base.hpp>
 
+#include <string>
 #include <utility>
 
+
 namespace mongocxx {
 MONGOCXX_INLINE_NAMESPACE_BEGIN
 namespace exception {
 
-base::base(bsoncxx::document::value raw_server_error)
-    : _raw_server_error(std::move(raw_server_error)) {
+base::base(std::error_code ec) : _ec{ec} {
+}
+
+base::base(std::error_code ec, std::string what_arg) : _ec{ec}, _what{std::move(what_arg)} {
 }
 
-base::base(error_and_code_type error_and_code) : _error_and_code(std::move(error_and_code)) {
+base::base(bsoncxx::document::value raw_server_error, std::error_code ec)
+    : _ec{ec}, _raw_server_error{std::move(raw_server_error)} {
 }
 
-base::base(bsoncxx::document::value raw_server_error, error_and_code_type error_and_code)
-    : _raw_server_error(std::move(raw_server_error)), _error_and_code(std::move(error_and_code)) {
+base::base(bsoncxx::document::value raw_server_error, std::error_code ec, std::string what_arg)
+    : _ec{ec}, _what{std::move(what_arg)}, _raw_server_error{std::move(raw_server_error)} {
```

Comment:
Please switch these so the initialization order matches the order in which these members are declared in the header file.